File: LongestPalindrome/LP.java
class Solution {
    public int longestPalindrome(String s) {
        HashMap<Character,Integer> map = new HashMap<Character,Integer>();
        for(int i=0;i<s.length();i++){
            char c = s.charAt(i);
            if(map.containsKey(c)) map.put(c,map.get(c)+1);
            else map.put(c,1);
        }
        int res = 0;
        for(Integer value:map.values()){
            res +=value - value%2;
            if(res%2==0 && value%2==1){
                res++;
            }
        }
        return res;
    }
}


class Solution {
    public int longestPalindrome(String s) {
        int[] arr = new int[128];
        int res = 0;
        for(int i=0;i<s.length();i++){
            arr[s.charAt(i)]++;
        }
        for(int c:arr){
            res+=c-c%2;
            if(res%2==0&&c%2==1) res++;
        }
        return res;
    }
}

File: LongestPalindrome/LP.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        num = list(collections.Counter(s).values())
        num1 = [0]+[n for n in num if n%2==0]
        num2 = [0]+[n-1 for n in num if n%2==1]
        res = 0
        if len(num2)>1:
            res=1
        return sum(num1)+sum(num2)+res


class Solution:
    def longestPalindrome(self, s: str) -> int:
        num = list(collections.Counter(s).values())
        res = 0
        for n in num:
            res+=n-n%2
            if n%2==1 and res%2==0:
                res+=1
        return res